include/pybind11/pytypes.h

```diff
@@ -558,6 +578,40 @@ struct error_fetch_and_normalize {
                     }
                 }
             }
+#if PY_VERSION_HEX >= 0x030B0000
+            auto notes
+                = reinterpret_steal<object>(PyObject_GetAttrString(m_value.ptr(), "__notes__"));
+            if (!notes) {
+                PyErr_Clear(); // No notes is good news.
+            } else {
+                auto len_notes = PyList_Size(notes.ptr());
+                if (len_notes < 0) {
+                    result += "\nFAILURE obtaining len(__notes__): " + detail::error_string();
+                } else {
+                    result += "\n__notes__ (len=" + std::to_string(len_notes) + "):";
+                    for (ssize_t i = 0; i < len_notes; i++) {
+                        PyObject *note = PyList_GET_ITEM(notes.ptr(), i);
+                        auto note_bytes = reinterpret_steal<object>(
```

Comment:
This is copy-pasted from the existing code further up (btw I didn't see enough meat to factor out).

https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_AsEncodedString

returns a new reference, we have to take ownership.

(Unfortuantely IMO) it's called "steal" instead of "take_ownership", I think this is correct.

Reply:
Oops, nvm. Misread this and didn't notice the PyUnicode_AsEncodedString. You are 100% right.